src/sagemaker/workflow/airflow.py

```diff
@@ -394,5 +392,223 @@ def model_config_from_estimator(instance_type, estimator, role=None, image=None,
     elif isinstance(estimator, sagemaker.estimator.Framework):
         model = estimator.create_model(model_server_workers=model_server_workers, role=role,
                                        vpc_config_override=vpc_config_override)
+    else:
+        raise TypeError('Estimator must be one of BYO estimator, framework estimator or amazon algorithm'
+                        'estimator.')
 
     return model_config(instance_type, model, role, image)
+
+
+def transform_config(transformer, data, data_type='S3Prefix', content_type=None, compression_type=None,
```

Comment:
Yep I do want it to be reused in session. Now I cannot because I need to bypass all the validations/describe calls/etc to use Jinja templating. 

If in session, we can construct the config first, and then apply all manipulations afterwards. That would be awesome and these two parts can be coupled. (which is really good since for now I am worried if session part got updated, for example, new entries in boto config introduced, airflow part cannot catch it)